To: vim-dev@vim.org
Subject: Patch 7.3.019
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.019
Problem:    ":nbstart" can fail silently.
Solution:   Give an error when netbeans is not supported by the GUI. (Xavier
	    de Gaye)
Files:	    src/netbeans.c


*** ../vim-7.3.018/src/netbeans.c	2010-09-29 17:26:57.000000000 +0200
--- src/netbeans.c	2010-09-29 17:35:37.000000000 +0200
***************
*** 2773,2778 ****
--- 2773,2788 ----
  ex_nbstart(eap)
      exarg_T	*eap;
  {
+ #ifdef FEAT_GUI
+ # if !defined(FEAT_GUI_X11) && !defined(FEAT_GUI_GTK)  \
+                 && !defined(FEAT_GUI_W32)
+     if (gui.in_use)
+     {
+         EMSG(_("E838: netbeans is not supported with this GUI"));
+         return;
+     }
+ # endif
+ #endif
      netbeans_open((char *)eap->arg, FALSE);
  }
  
*** ../vim-7.3.018/src/version.c	2010-09-29 18:42:25.000000000 +0200
--- src/version.c	2010-09-30 21:02:03.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     19,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
233. You start dreaming about web pages...in html.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
